### ISSUE TITLE
Make the Curve() operator term more user-friendly.

### ISSUE THREAD

**Issue by krivit** [statnet/ergm#248]
- [ ] Allow `map` and `gradient` arguments to be specified as `purrr`-style formulas.
- [x] Allow the user to flag the `gradient` as "linear", in which case its (constant) value will be computed by calling `map` repeatedly.
- [x] Implement some shortcuts for common use cases of `map` as well? E.g., "replicate" to simply replicate the input parameter.
- [ ] Allow the user to request numerical differentiation to be used in evaluating `gradient`.
- [ ] Rename and/or alias the term to `Parametrize()`?
- [ ] Add an option to replace the existing curved ERGM structure of the formula with that specified?

**Comment by krivit** [comment]
@drh20drh20 , @martinamorris , what do you think about the "Rename and/or alias the term to Parametrize()" item?

**Comment by martinamorris** [comment]
Hmm.  It's long.  Since I haven't really dug into what this operator does, I'm not sure I have a good sense of what it should be called.

**Comment by drh20drh20** [comment]
I agree that `Parametrize` is more descriptive than `Curve`.  Perhaps this is an(other) example in which we can add a second operator.  In this case, `Parametrize` is a synonym for `Curve`, which would need to be kept around for reverse compatibility only, and presumably only for a period of time.
